packages/forms/backend/sirius-components-forms-graphql/src/main/java/org/eclipse/sirius/components/forms/graphql/datafetchers/form/SelectOptionIconURLDataFetcher.java

```diff
@@ -25,17 +27,26 @@
  * If the <code>SelectOption.iconURL</code> is of the form <code>path/to/image.svg</code>, the rewritten value which will
  * be seen by the frontend will be <code>/api/images/path/to/image.svg</code>.
  *
+ * If the <code>SelectOption.iconURL</code> is of the form <code>path/to/UUID.svg</code>, the rewritten value which will
+ * be seen by the frontend will be <code>/api/images/custom/path/to/image.svg</code>.
+ *
  * @author mcharfadi
  */
 @QueryDataFetcher(type = "SelectOption", field = "iconURL")
 public class SelectOptionIconURLDataFetcher implements IDataFetcherWithFieldCoordinates<String> {
-
+    private static final Pattern UUID_REGEX =
+            Pattern.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
     @Override
     public String get(DataFetchingEnvironment environment) throws Exception {
         SelectOption option = environment.getSource();
         String result = option.getIconURL();
         if (result != null && !result.isBlank()) {
-            result = URLConstants.IMAGE_BASE_PATH + result;
+            if (UUID_REGEX.matcher(result).matches()) {
```

Comment:
`CustomImageMetadata` already has a `getURL` which includes the `/custom` part.
In `NodeStylePropertiesConfigurer#createIconSelectionField`, you can use that directly instead of setting the "iconURL" to the image's UUID and then trying to parse and adapt it jhere.